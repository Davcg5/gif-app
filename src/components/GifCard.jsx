import React from 'react'

export default function GifCard({id, title, url}) {


    return (
    <div className='card'>
        <img src={url} alt={title} />
    </div>
  )
}
