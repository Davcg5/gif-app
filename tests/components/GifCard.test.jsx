import { render, screen } from "@testing-library/react"
import GifCard from "../../src/components/GifCard"

describe('Tests in GifCard', () => {
    const title = 'Goku'
    const url = 'https://goku.com/'  
    test('should match the snapshot', () => { 
    
        const {container} = render (
            <GifCard title={title} url = {url}/>
        )
        expect(container).toMatchSnapshot();
    })

    test('should show image with url and said alt', () => { 
        render (
            <GifCard title={title} url = {url}/>
        )
        //screen.debug()

        const {alt, src} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
    })

    test('should show title in the component', () => { 
        render (
            <GifCard title={title} url = {url}/>
        )
        expect(screen.getByText(title)).toBeTruthy();
    })
})