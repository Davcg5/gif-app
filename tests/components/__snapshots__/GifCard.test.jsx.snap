// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifCard should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://goku.com/"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
